Clean up and unify CI configs

This repo has both Travis and Google Cloud Builds CI configurations, but there are some differences:

1) Travis uses `-race` when it runs the tests, cloudbuilds does not.
2) Cloudbuilds uses a different shell snippet to check for 100% coverage than travis uses. They should use the same.

`-race` was not used on cloudbuilds because it would fail:
```
Step #1 - "test": Running: go test -v -race -coverprofile=coverage.out -covermode=atomic ./...
Step #1 - "test": # runtime/race
Step #1 - "test": race_linux_amd64.syso: In function `__sanitizer::GetArgv()':
Step #1 - "test": gotsan.cc:(.text+0x4183): undefined reference to `__libc_stack_end'
Step #1 - "test": race_linux_amd64.syso: In function `__sanitizer::ReExec()':
Step #1 - "test": gotsan.cc:(.text+0x9797): undefined reference to `__libc_stack_end'
Step #1 - "test": race_linux_amd64.syso: In function `__sanitizer::InternalAlloc(unsigned long, __sanitizer::SizeClassAllocatorLocalCache<__sanitizer::SizeClassAllocator32<__sanitizer::AP32> >*, unsigned long)':
Step #1 - "test": gotsan.cc:(.text+0xaac1): undefined reference to `__libc_malloc'
Step #1 - "test": race_linux_amd64.syso: In function `__sanitizer::InternalRealloc(void*, unsigned long, __sanitizer::SizeClassAllocatorLocalCache<__sanitizer::SizeClassAllocator32<__sanitizer::AP32> >*)':
Step #1 - "test": gotsan.cc:(.text+0xca20): undefined reference to `__libc_realloc'
Step #1 - "test": race_linux_amd64.syso: In function `__sanitizer::InternalFree(void*, __sanitizer::SizeClassAllocatorLocalCache<__sanitizer::SizeClassAllocator32<__sanitizer::AP32> >*)':
Step #1 - "test": gotsan.cc:(.text+0x66e8): undefined reference to `__libc_free'
Step #1 - "test": collect2: error: ld returned 1 exit status
```

That is because the cloud-builder is an alpine image on which the race detector does not work because alpine is not glibc-based.

To make that work, we'll need to switch to a debian-based go cloud-builder.

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,7 +7,7 @@ before_script:
   - curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | sh -s -- -b $(go env GOPATH)/bin v1.13.2
 script:
   - golangci-lint run
-  - go test -v -race  -coverprofile=coverage.txt -covermode=atomic ./...
-  - go tool cover -func=coverage.txt | awk '/^total:/ { gsub(/%$/, ""); if ($3 < 100.0) {print "FAIL - Coverage below 100%"; exit(1)} }'
+  - go test -v -race -coverprofile=coverage.out -covermode=atomic ./...
+  - "{ go tool cover -func=coverage.out | grep '^total:.*100.0%'; } || { echo 'FAIL - Coverage below 100%'; exit 1; }"
 after_success:
   - bash <(curl -s https://codecov.io/bash)

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -1,14 +1,14 @@
 steps:
   - id: version
-    name: 'gcr.io/cloud-builders/go'
+    name: 'gcr.io/cloud-builders/go:debian'
     args: ['version']
     env: ['GOPATH=/workspace/_gopath', 'GO111MODULE=on']
   - id: test
-    name: 'gcr.io/cloud-builders/go'
-    args: ['test', '-v', '-coverprofile=coverage.out', '-covermode=atomic' , './...']
+    name: 'gcr.io/cloud-builders/go:debian'
+    args: ['test', '-v', '-race', '-coverprofile=coverage.out', '-covermode=atomic' , './...']
     env: ['GOPATH=/workspace/_gopath', 'GO111MODULE=on']
   - id: cover
-    name: 'gcr.io/cloud-builders/go'
+    name: 'gcr.io/cloud-builders/go:debian'
     env: ['GOPATH=/workspace/_gopath', 'GO111MODULE=on']
     entrypoint: 'sh'
     args:

--- a/cloudbuild.yaml
+++ b/cloudbuild.yaml
@@ -13,4 +13,4 @@ steps:
     entrypoint: 'sh'
     args:
       - -c 
-      - "{ go tool cover -func=coverage.out | grep -s '^total:.*100.0%'; } || { echo 'FAIL - Coverage below 100%'; exit 1; }"
+      - "{ go tool cover -func=coverage.out | grep '^total:.*100.0%'; } || { echo 'FAIL - Coverage below 100%'; exit 1; }"

